include/boost/hana/experimental/printable.hpp

```diff
@@ -111,6 +111,18 @@ BOOST_HANA_NAMESPACE_BEGIN namespace experimental {
         }
     };
 
+    // model for hana::optional
+    template <>
+    struct print_impl<hana::optional_tag> {
+        template <typename O>
+        static auto apply(O const& optional) {
+            return hana::maybe("nothing",
+                [](auto x) {
+                    return "just " + hana::experimental::print(x);
```

Comment:
I was on the fence about it. I almost did `Just 5` to be like haskell, though I am not a haskell programmer.